metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/configuration/MetadataImporterMainApplicationConfiguration.java

```diff
@@ -47,34 +51,59 @@
-                studyConverter(), publicationExtractorFromStudy(publicationRepository),
-                webResourceExtractorFromStudy(webResourceRepository), taxonomyExtractor(taxonomyRepository),
-                sraAnalysisXmlParser(), analysisConverter(), fileExtractorFromAnalysis(fileRepository),
-                analysisRepository, studyRepository);
+                                                   PublicationRepository publicationRepository,
+                                                   WebResourceRepository webResourceRepository,
+                                                   FileRepository fileRepository,
+                                                   TaxonomyRepository taxonomyRepository,
+                                                   ReferenceSequenceRepository referenceSequenceRepository,
+                                                   AnalysisRepository analysisRepository,
+                                                   StudyRepository studyRepository) {
+        return new SraObjectsImporterThroughAPI(
+                sraXmlRetrieverThroughApi,
+                sraStudyXmlParser(),
+                studyConverter(),
+                publicationExtractorFromStudy(publicationRepository),
+                webResourceExtractorFromStudy(webResourceRepository),
+                taxonomyExtractor(taxonomyRepository),
+                taxonomyExtractorFromReferenceSequence(taxonomyRepository),
+                sraAnalysisXmlParser(),
+                analysisConverter(),
+                fileExtractorFromAnalysis(fileRepository),
+                sraAssemblyXmlParser(),
+                referenceSequenceConverter(),
+                analysisRepository,
+                studyRepository,
+                referenceSequenceRepository
+        );
     }
 
     @Bean
     @ConditionalOnProperty(name = "import.source", havingValue = "DB")
-    public ObjectsImporter objectImporterThroughEnaDatabase(
-            SraXmlRetrieverThroughDatabase sraXmlRetrieverThroughDatabase,
-            PublicationRepository publicationRepository,
-            WebResourceRepository webResourceRepository,
-            FileRepository fileRepository,
-            TaxonomyRepository taxonomyRepository,
-            AnalysisRepository analysisRepository,
-            StudyRepository studyRepository) {
-        return new SraObjectsImporterThroughDatabase(sraXmlRetrieverThroughDatabase, sraStudyXmlParser(),
-                studyConverter(), publicationExtractorFromStudy(publicationRepository),
-                webResourceExtractorFromStudy(webResourceRepository), taxonomyExtractor(taxonomyRepository),
-                sraAnalysisXmlParser(), analysisConverter(), fileExtractorFromAnalysis(fileRepository),
-                analysisRepository,studyRepository);
+    public ObjectsImporter objectImporterThroughEnaDatabase(SraXmlRetrieverThroughDatabase sraXmlRetrieverThroughDatabase,
```

Comment:
SraXmlRetrieverThroughDatabase can be moved to the next line.

Reply:
Fixed, thank you